beacon-chain/blockchain/service.go

```diff
@@ -195,43 +189,26 @@ func (c *ChainService) updateHead(processedBlock <-chan *types.Block) {
 			}
 
 			// TODO(#674): Handle chain reorgs.
-			var newCState *types.CrystallizedState
-			if c.unfinalizedBlocks[h].cycleTransition {
-				newCState = blockcState
+			var newState *types.BeaconState
+			if c.unfinalizedBlocks[h].IsCycleTransition(block.SlotNumber()) {
```

Comment:
Now that the state's updated every slot, this if statement shouldn't be necessary.